From bff534413fa6a09e8363b914ca8f12df73c93b2c Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Tue, 8 Mar 2022 11:27:40 +0100
Subject: [PATCH 14/17] wpa_supplicant: add prefix wpa_ to crypto functions

Prefix `_wpa` added to further crypto functions of `wpa_suppplicant` to avoid name conflicts with RIOT modules `crypto` and `hashes`.
---
 components/wpa_supplicant/src/crypto/aes-internal-dec.c | 2 +-
 components/wpa_supplicant/src/crypto/aes-internal-enc.c | 2 +-
 components/wpa_supplicant/src/crypto/aes-internal.c     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/components/wpa_supplicant/src/crypto/aes-internal-dec.c b/components/wpa_supplicant/src/crypto/aes-internal-dec.c
index f72742cf590..70a40696943 100644
--- a/components/wpa_supplicant/src/crypto/aes-internal-dec.c
+++ b/components/wpa_supplicant/src/crypto/aes-internal-dec.c
@@ -31,7 +31,7 @@ static int rijndaelKeySetupDec(u32 rk[], const u8 cipherKey[], int keyBits)
 	u32 temp;
 
 	/* expand the cipher key: */
-	Nr = rijndaelKeySetupEnc(rk, cipherKey, keyBits);
+	Nr = wpa_rijndaelKeySetupEnc(rk, cipherKey, keyBits);
 	if (Nr < 0)
 		return Nr;
 	/* invert the order of the round keys: */
diff --git a/components/wpa_supplicant/src/crypto/aes-internal-enc.c b/components/wpa_supplicant/src/crypto/aes-internal-enc.c
index a856dc9f3f8..1bedc885368 100644
--- a/components/wpa_supplicant/src/crypto/aes-internal-enc.c
+++ b/components/wpa_supplicant/src/crypto/aes-internal-enc.c
@@ -106,7 +106,7 @@ void * aes_encrypt_init(const u8 *key, size_t len)
 	rk = os_malloc(AES_PRIV_SIZE);
 	if (rk == NULL)
 		return NULL;
-	res = rijndaelKeySetupEnc(rk, key, len * 8);
+	res = wpa_rijndaelKeySetupEnc(rk, key, len * 8);
 	if (res < 0) {
 		os_free(rk);
 		return NULL;
diff --git a/components/wpa_supplicant/src/crypto/aes-internal.c b/components/wpa_supplicant/src/crypto/aes-internal.c
index bd4535d2096..6b646027c8c 100644
--- a/components/wpa_supplicant/src/crypto/aes-internal.c
+++ b/components/wpa_supplicant/src/crypto/aes-internal.c
@@ -776,7 +776,7 @@ const u8 rcons[] = {
  *
  * @return	the number of rounds for the given cipher key size.
  */
-int rijndaelKeySetupEnc(u32 rk[], const u8 cipherKey[], int keyBits)
+int wpa_rijndaelKeySetupEnc(u32 rk[], const u8 cipherKey[], int keyBits)
 {
 	int i;
 	u32 temp;
-- 
2.17.1

